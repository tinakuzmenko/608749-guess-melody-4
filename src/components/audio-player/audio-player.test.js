import React from "react";
import renderer from "react-test-renderer";
import AudioPlayer from "./audio-player.jsx";

const mock = {
  song: {
    src: `https://commons.wikimedia.org/wiki/File:Egmont_Overture_Finale_(ISRC_USUAN1200070).mp3`
  }
};

describe(`AudioPlayer`, () => {
  it(`Renders correctly when not playing`, () => {
    const tree = renderer
      .create(<AudioPlayer
        isLoading={true}
        isPlaying={false}
        onPlayButtonClick={() => {}}
        src={mock.song.src}
      >
        <audio />
      </AudioPlayer>, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Renders correctly when playing`, () => {
    const tree = renderer
      .create(<AudioPlayer
        isLoading={false}
        isPlaying={true}
        onPlayButtonClick={() => {}}
        src={mock.song.src}
      >
        <audio />
      </AudioPlayer>, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
