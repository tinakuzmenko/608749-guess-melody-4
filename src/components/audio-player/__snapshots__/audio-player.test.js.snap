// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer Renders correctly when not playing 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;

exports[`AudioPlayer Renders correctly when playing 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={false}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
